error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ExplicitArgumentEnumeration.java

```diff
@@ -0,0 +1,200 @@
+                m -> isPublic(m) && !m.equals(method),
+                method.enclClass().type,
+                state.getTypes())
+            .collect(toImmutableList());
+
+    /*
+     * If all overloads have a single parameter, and at least one of them is a varargs method, then
+     * we assume that unwrapping the iterable argument will cause a suitable overload to be invoked.
+     * (Note that there may be multiple varargs overloads, either with different parameter types, or
+     * due to method overriding; this check is does not attempt to determine which exact method or
+     * overload will be invoked as a result of the suggested simplification.)
+     *
+     * Note that this is a (highly!) imperfect heuristic, but it is sufficient to prevent e.g.
+     * unwrapping of arguments to `org.jooq.impl.DSL#row`, which can cause the expression's return
+     * type to change from `RowN` to (e.g.) `Row2`.
+     */
+    // XXX: There are certainly cases where it _would_ be nice to unwrap the arguments to
+    // `org.jooq.impl.DSL#row(Collection<?>)`. Look into this.
+    // XXX: Ideally we do check that one of the overloads accepts the unwrapped arguments.
+    // XXX: Ideally we validate that eligible overloads have compatible return types.
+    boolean hasLikelySuitableVarargsOverload =
+        overloads.stream().allMatch(m -> m.params().size() == 1)
+            && overloads.stream().anyMatch(MethodSymbol::isVarArgs);
+
+    return hasLikelySuitableVarargsOverload
+        ? Optional.of(SourceCode.unwrapMethodInvocation(argument, state))
+        : Optional.empty();
+  }
+
+  private static Optional<SuggestedFix> trySuggestCallingCustomAlternative(
+      MethodInvocationTree tree, MethodInvocationTree argument, VisitorState state) {
+    return ALTERNATIVE_METHODS.rowMap().entrySet().stream()
+        .filter(e -> e.getKey().matches(tree, state))
+        .findFirst()
+        .flatMap(e -> trySuggestCallingCustomAlternative(tree, argument, state, e.getValue()));
+  }
+
+  private static Optional<SuggestedFix> trySuggestCallingCustomAlternative(
+      MethodInvocationTree tree,
+      MethodInvocationTree argument,
+      VisitorState state,
+      Map<String, String> alternatives) {
+    String name = ASTHelpers.getSymbol(tree).getSimpleName().toString();
+    String alternative = alternatives.get(name);
+
+    if (alternative == null) {
+      return Optional.empty();
+    }
+
+    SuggestedFix fix = SourceCode.unwrapMethodInvocation(argument, state);
+    return Optional.of(
+        alternative.equals(name)
+            ? fix
+            : SuggestedFix.builder()
+                .merge(SuggestedFixes.renameMethodInvocation(tree, alternative, state))
+                .merge(fix)
+                .build());
+  }
+
+  // XXX: Once we target JDK 14+, drop this method in favour of `Symbol#isPublic()`.
+  private static boolean isPublic(Symbol symbol) {
+    return symbol.getModifiers().contains(Modifier.PUBLIC);
+  }
```

Comment:
Also present in #972, but a separate utility method doesn't seem worth it. ([Or we can just drop JDK 11 support](https://github.com/PicnicSupermarket/error-prone-support/pull/603#issuecomment-1902727809) :upside_down_face:.)